test/e2e/wireguard_test.go

```diff
@@ -127,3 +129,62 @@ func testServiceConnectivity(t *testing.T, data *TestData) {
 	err = data.runNetcatCommandFromTestPod(clientPodName, testNamespace, clientPodNodeIP, svc.Spec.Ports[0].NodePort)
 	require.NoError(t, err, "Pod %s should be able to connect the service's NodePort %s:%d, but was not able to connect", clientPodName, clientPodNodeIP, svc.Spec.Ports[0].NodePort)
 }
+
+// testRoute make sure that the route to Pod on peer Node or route to peer gateway is targeting the WireGuard device.
+func testRoute(t *testing.T, data *TestData) {
+	srcNode := nodeName(0)
+	var srcNodeIP, peerGatewayIP, peerPodIP string
+	IPv4, IPv6 := nodeGatewayIPs(0)
+	if IPv4 != "" {
+		srcNodeIP = IPv4
+	} else {
+		srcNodeIP = IPv6
+	}
+	IPv4, IPv6 = nodeGatewayIPs(1)
+	if IPv4 != "" {
+		peerGatewayIP = IPv4
+	} else {
+		peerGatewayIP = IPv6
+	}
+
+	peerPod := randName("agnhost-")
+	require.NoError(t, data.createAgnhostPodOnNode(peerPod, testNamespace, nodeName(1), false))
+	defer data.deletePodAndWait(defaultTimeout, peerPod, testNamespace)
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, peerPod, testNamespace))
+
+	peerPodIPs, err := data.podWaitForIPs(defaultTimeout, peerPod, testNamespace)
+	require.NoError(t, err)
+	if peerPodIPs.ipv4 != nil {
+		peerPodIP = peerPodIPs.ipv4.String()
+	} else {
+		peerPodIP = peerPodIPs.ipv6.String()
+	}
+
+	tests := []struct {
+		name               string
+		dstIP              string
+		expectedDeviceName string
+		expectedSrcIP      string
+	}{
+		{
+			name:               "toPodOnPeerNode",
+			dstIP:              peerPodIP,
+			expectedDeviceName: "antrea-wg0",
+			expectedSrcIP:      srcNodeIP,
+		},
+		{
+			name:               "toPeerGateway",
+			dstIP:              peerGatewayIP,
+			expectedDeviceName: "antrea-wg0",
+			expectedSrcIP:      srcNodeIP,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cmd := fmt.Sprintf("ip route get %v", tt.dstIP)
+			_, stdout, _, err := data.RunCommandOnNode(srcNode, cmd)
+			require.NoError(t, err)
+			assert.Contains(t, stdout, tt.expectedDeviceName, tt.expectedSrcIP)
```

Comment:
tt.expectedSrcIP is not being checked here.
```suggestion
			assert.Contains(t, stdout, tt.expectedDeviceName)
                        assert.Contains(t, stdout, tt.expectedSrcIP)
```